tree/ntuple/v7/src/RField.cxx

```diff
@@ -1083,6 +1083,211 @@ void ROOT::Experimental::RVectorField::AcceptVisitor(Detail::RFieldVisitor &visi
+{
+   void **rvecAddr = reinterpret_cast<void**>(value.GetRawPtr());
+   // void* fBegin is the first data member
+   void *begin = *rvecAddr;
+   // int32_t fSize is the second data member (after 1 void*, no padding expected)
+   const int32_t* sizePtr = reinterpret_cast<int32_t *>(rvecAddr + 1);
+   const int32_t size = *sizePtr;
+   R__ASSERT(size >= 0);
+   // int32_t fCapacity is the third data member (after fSize, no padding expected)
+   const int32_t capacity = *(sizePtr + 1);
+
+   for (int32_t i = 0; i < size; ++i) {
+      auto elementValue = fSubFields[0]->CaptureValue(reinterpret_cast<void *>(std::intptr_t(begin) + i * fItemSize));
+      fSubFields[0]->DestroyValue(elementValue, true /* dtorOnly */);
+   }
+
+   // figure out if we are in the small state, i.e. begin == &inlineBuffer
+   // there might be padding between fCapacity and the inline buffer, so we compute it here
+   const auto dataMemberSz = sizeof(void *) + 2 * sizeof(int32_t);
+   const auto alignOfT = fSubFields[0]->GetAlignment();
+   auto paddingMiddle = dataMemberSz % alignOfT;
+   if (paddingMiddle != 0)
+      paddingMiddle = alignOfT - paddingMiddle;
+   const bool isSmall = begin == (rvecAddr + dataMemberSz + paddingMiddle);
+
+   const bool owns = capacity != -1;
+   if (!dtorOnly && !isSmall && owns)
+      free(begin);
+}
+
+ROOT::Experimental::Detail::RFieldValue ROOT::Experimental::RRVecField::CaptureValue(void* where)
+{
+   return Detail::RFieldValue(true /* captureFlag */, this, where);
+}
+
+std::vector<ROOT::Experimental::Detail::RFieldValue>
+ROOT::Experimental::RRVecField::SplitValue(const Detail::RFieldValue &value) const
+{
+   void **rvecAddr = reinterpret_cast<void**>(value.GetRawPtr());
+   // void* fBegin is the first data member
+   void *begin = *rvecAddr;
+   // int32_t fSize is the second data member (after 1 void*)
+   const int32_t size = *reinterpret_cast<int32_t *>(rvecAddr + 1);
+   R__ASSERT(size >= 0);
+
+   std::vector<Detail::RFieldValue> result;
+   for (int32_t i = 0; i < size; ++i) {
+      auto elementValue = fSubFields[0]->CaptureValue(reinterpret_cast<void *>(std::intptr_t(begin) + i * fItemSize));
+      result.emplace_back(std::move(elementValue));
+   }
+   return result;
+}
+
+size_t ROOT::Experimental::RRVecField::GetValueSize() const
+{
+   // the size of an RVec<T> is the size of its 4 data-members + optional padding:
+   //
+   // data members:
+   // - void *fBegin
+   // - int32_t fSize
+   // - int32_t fCapacity
+   // - the char[] inline storage, which is aligned like T
+   //
+   // padding might be present:
+   // - between fCapacity and the char[] buffer aligned like T
+   // - after the char[] buffer
+
+   const auto dataMemberSz = sizeof(void *) + 2 * sizeof(int32_t);
+   const auto alignOfT = fSubFields[0]->GetAlignment();
+   const auto sizeOfT = fSubFields[0]->GetValueSize();
+
+   // mimic the logic of RVecInlineStorageSize, but at runtime
+   const auto inlineStorageSz = [&] {
```

Comment:
Not sure why we need a lambda here.

Reply:
It's the IIFE pattern, e.g. https://www.bfilipek.com/2016/11/iife-for-complex-initialization.html?m=1

It's to clearly distinguish the not-so-trivial logic that's only there to initialize this variable.
I can just put everything inline if you prefer